### PR TITLE
Migrate Stock Panel to Homescreen

### LINKED ISSUE #5239: Home Screen - migrate Stock panel
**User story:**
As a merchant
I want to easily see how many products are low stock or out of stock when I enter WooCommerce
So I can restock inventory as soon as possible

**Description:** we’re going to migrate the stock panel features to the home screen, as managing inventory is one of the most important tasks for stores with Stock Management enabled

**Acceptance criteria:**
- [ ] Display a Stock task in the home screen in stores with [Stock Management](https://docs.woocommerce.com/document/managing-products/#inventory-tab) enabled
- [ ] Display up to 5 products that are either Low in Stock or Out of Stock

- [ ] The product card should include the same elements that we have now: image, product name, current stock and `Update Stock` CTA. Ideally we should also add the last time that a product was ordered (nice to have)
- [ ] The user is redirected to the product page when clicking in the product name
- [ ] Clicking in the `Update Stock` CTA opens a modal allowing the user to update the product stock without leaving the home screen
 
- [ ] The stock activity panel should be removed from the navbar once we have the same feature in the home screen

**Event tracking:**
Can we reuse the same event tracking that we have in the stock panel?

**Additional links:**
[P2 with links to the Figma file](pbIJXs-ma-p2)

### PR DESCRIPTION
**Pull request by jeffstieler**
Fixes #5239.

This PR primarily relocates the Stock Panel to the Homescreen.

Several supporting changes were made:

* Updated the `<ProductImage>` styling to match designs (affects Reviews panel as well)
* Added "last order date" to the `/wc-analytics/products` endpoint response
* Let `<AccordionPanel>` prop `initialOpen` toggle the panel if it hasn't been interacted with by the user (allows a panel to be closed while data fetching is occurring, if desired)
* Enhanced the "items" data store to update total counts whenever identical queries (from a totals perspective) are issued - allows for the Panel `count` to change without issuing another query specifically for the total count, since that header will be returned in the query for more items

### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots

Screen Shot 2020-11-24 at 9 56 33 AM

### Detailed test instructions:

- Ensure your test store has inventory management enabled
- With no products out of stock, verify the panel says "0" and is closed
- With less than 5 products out of stock, verify the panel shows the exact number of placeholders while loading
- With more than 5 products out of stock, verify the panel shows only 5 placeholders while loading
- (Still with 5+) Update a product so it is no longer out of stock
- Verify the product dismisses and another is fetched
- Verify the count decreases
- Update another product
- Click "undo" on the snackbar notice
- Verify the changes were undone
- Verify the stock panel is no longer in the header

### Changelog Note:

Enhancement: move Stock management to the home screen.

### CODE REVIEW

File: client/homescreen/activity-panel/stock/index.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	}
+
+	async updateStock( product, quantity ) {
+		const { invalidateResolution, updateProductStock } = this.props;
+
+		const result = await updateProductStock( product, quantity );
+
+		if ( result.success ) {
+			// Request more low stock products.
+			invalidateResolution( 'getItems', [ 'products', productsQuery ] );
```
Comment (louwie17) [inline client/homescreen/activity-panel/stock/index.js:49]: I did notice the same issue I had with the `reviews` that if you update the stock when there is only one item, and then click `undo` it won't update the homescreen (it did correctly revert the stock).

Although I can't reproduce this every-time only on a fresh page load, and it did eventually update after expanding/minimizing the panel.

I had to invalidate the `getItemsTotalCount` in that case to re-trigger the search.
update-stock

**Review by louwie17 — APPROVED** [review]
Good stuff @jeffstieler, also thanks for getting to some of that Activity Panel styling.
It tested and worked great! Just left two notes, but they aren't blockers in my opinion, so you got my stamp of approval 👊 💯 

🤞 I won't have to many merge conflicts 🙈

File: client/homescreen/activity-panel/stock/index.js (hunk 1 above)
Comment (jeffstieler) [reply client/homescreen/activity-panel/stock/index.js:49]: I've mitigated this somewhat in https://github.com/woocommerce/woocommerce-admin/pull/5729/commits/d48edc2946faafc9b9e792527d2d9715b94e5af0 by defaulting the panel to closed (as per Slack convo). If you close the panel before clicking "undo" in the notice, the panel won't refresh - that's an edge case that I think can be handled in a follow up PR.

File: client/homescreen/activity-panel/stock/card.js [hunk 2]
```diff
@@ -78,7 +76,7 @@ class ProductStockCard extends Component {
 
 	async onSubmit() {
 		const { product, updateProductStock, createNotice } = this.props;
-		const { quantity } = this.state;
+		const quantity = parseInt( this.state.quantity, 10 );
 
 		this.setState( { editing: false, edited: true } );
```
[the remark this answers is not in the document]
Comment (jeffstieler) [reply client/homescreen/activity-panel/stock/card.js:88]: Simple enough - done in a1fb1af.

**Pull request merged**